### ISSUE TITLE
AdamW has not set 'maximize' flag properly to adamw in its step method

### ISSUE THREAD

**Issue by chris-opendata** [pytorch/pytorch#109992]
### 🐛 Describe the bug

I set `maximize` to `True` in `AdamW` constructor as I use it as a maximizer. When stepping into its `step()` method during debugging, I found that the `group["maximize"]` to `F.adamw` function is `False`. Meanwhile, AdamW's `defaults` has `maximize` `True`.
It means that the argument set in the constructor is recorded in the `defaults` but it was not properly passed onto `param_groups`.

### Versions

absl-py==0.14.0
accelerate==0.9.0
-e git+https://github.com/acl-org/aclpubcheck.git@7db49859de9728837a4c3f70008aac2c84d35784#egg=aclpubcheck
aiohttp==3.8.0
aiosignal==1.2.0
alembic==1.8.1
aniso8601==9.0.1
antlr4-python3-runtime==4.8
antu==0.0.5a0
argon2-cffi==20.1.0
ase==3.21.1
asttokens==2.0.5
async-generator==1.10
async-timeout==4.0.0
attrs==21.2.0
autopage==0.5.1
backcall==0.2.0
bibtexparser==1.4.0
bidict==0.21.3
bitarray==2.3.4
bleach==3.3.0
blis==0.7.6
cachetools==4.2.2
catalogue==2.0.6
certifi==2021.5.30
cffi==1.14.5
chardet==4.0.0
charset-normalizer==2.0.7
class-resolver==0.3.10
click==8.0.1
click-default-group==1.2.2
cliff==3.10.1
cmaes==0.8.2
cmd2==2.4.2
colorama==0.4.4
colorlog==6.6.0
cryptography==38.0.2
cycler==0.10.0
cymem==2.0.6
Cython==0.29.23
dataclasses==0.6
dataclasses-json==0.5.7
datasets==1.15.1
decorator==4.4.2
defusedxml==0.7.1
dgl-cu111==0.6.1
dill==0.3.4
docdata==0.0.3
dyNET==2.1.2
en-core-web-lg @ https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.2.0/en_core_web_lg-3.2.0-py3-none-any.whl
entrypoints==0.3
executing==0.8.3
filelock==3.0.12
filetype==1.0.10
frozenlist==1.2.0
fsspec==2021.11.0
future==0.18.2
fuzzywuzzy==0.18.0
gensim==4.1.0
google-auth==1.35.0
google-auth-oauthlib==0.4.6
googledrivedownloader==0.4
graphene==3.0
graphql-core==3.1.6
graphql-relay==3.1.0
graphviz==0.19.1
greenlet==1.1.2
grpcio==1.41.0
h5py==3.2.1
huggingface-hub==0.1.2
hydra-core==1.1.1
icecream==2.1.2
idna==2.10
imageio==2.9.0
importlib-metadata==4.8.1
importlib-resources==5.2.2
ipykernel==5.5.5
ipython==7.24.1
ipython-genutils==0.2.0
ipywidgets==7.6.3
isodate==0.6.0
jedi==0.18.0
Jinja2==3.0.1
joblib==1.0.1
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.12
jupyter-console==6.4.0
jupyter-core==4.7.1
jupyterlab-pygments==0.1.2
jupyterlab-widgets==1.0.0
kiwisolver==1.3.1
langcodes==3.3.0
latexcodec==2.0.1
lightning-utilities==0.9.0
littleutils==0.2.2
llvmlite==0.36.0
lxml==4.9.1
Mako==1.2.1
Markdown==3.3.4
MarkupSafe==2.0.1
marshmallow==3.17.0
marshmallow-enum==1.5.1
matplotlib==3.4.2
matplotlib-inline==0.1.2
mistune==0.8.4
more-click==0.1.1
more-itertools==8.13.0
multidict==5.2.0
multiprocess==0.70.12.2
murmurhash==1.0.6
mypy-extensions==0.4.3
nbclient==0.5.3
nbconvert==6.0.7
nbformat==5.1.3
nest-asyncio==1.5.1
networkx==2.7.1
nltk==3.6.2
notebook==6.4.0
numba==0.53.1
numpy==1.20.3
oauthlib==3.1.1
ogb==1.3.1
omegaconf==2.1.1
optuna==2.10.1
outdated==0.2.1
overrides==6.1.0
packaging==20.9
pandas==1.3.5
pandocfilters==1.4.3
parso==0.8.2
pathy==0.6.1
pbr==5.9.0
pdfminer.six==20220524
pdfplumber==0.7.5
pickleshare==0.7.5
Pillow==9.2.0
portalocker==2.3.2
preshed==3.0.6
prettytable==3.3.0
prometheus-client==0.11.0
prompt-toolkit==3.0.18
protobuf==3.18.0
psutil==5.8.0
pyarrow==6.0.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pybtex==0.24.0
pycparser==2.20
pydantic==1.8.2
pyDeprecate==0.3.1
Pygments==2.9.0
pykeen==1.9.0
pylatexenc==2.10
pyparsing==2.4.7
pyperclip==1.8.2
pyreadline3==3.4.1
pyrsistent==0.17.3
pystow==0.4.6
pytextrank==3.2.3
python-dateutil==2.8.1
python-igraph==0.9.6
python-louvain==0.15
pythonds==1.2.1
pytorch-lightning==1.5.3
pytz==2021.1
PyWavelets==1.1.1
pywin32==301
pywinpty==1.1.2
PyYAML==5.4.1
pyzmq==22.1.0
qtconsole==5.1.0
QtPy==1.9.0
rapidfuzz==3.1.1
rdflib==5.0.0
rebiber==1.1.3
regex==2021.7.6
requests==2.25.1
requests-oauthlib==1.3.0
rexmex==0.1.0
rouge-score==0.0.4
rsa==4.7.2
sacrebleu==2.0.0
sacremoses==0.0.45
scikit-image==0.18.1
scikit-learn==0.24.2
scipy==1.8.0
Send2Trash==1.5.0
sentencepiece==0.1.96
simplejson==3.17.6
six==1.16.0
sklearn==0.0
smart-open==5.2.1
spacy==3.2.3
spacy-legacy==3.0.9
spacy-loggers==1.0.1
SQLAlchemy==1.4.27
srsly==2.4.2
stanfordcorenlp==3.9.1.1
stevedore==4.0.0
svgutils==0.3.4
tabulate==0.8.9
tensorboard==2.6.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.0
termcolor==2.0.1
terminado==0.10.1
testpath==0.5.0
texttable==1.6.4
thinc==8.0.13
threadpoolctl==2.1.0
tifffile==2021.6.6
tokenizers==0.10.3
torch==1.11.0+cu113
torch-cluster==1.6.0
torch-geometric==2.0.4
torch-max-mem==0.0.2
torch-ppr==0.0.8
torch-scatter==2.0.9
torch-sparse==0.6.13
torch-spline-conv==1.2.1
torchaudio==0.11.0+cu113
torchmetrics==1.0.1
torchtext==0.12.0
torchvision==0.12.0+cu113
tornado==6.1
tqdm==4.62.3
traitlets==5.0.5
transformers==4.9.0
tsv==1.2
typer==0.4.0
typing-extensions==3.10.0.0
typing-inspect==0.7.1
typing-utils==0.1.0
Unidecode==1.3.6
urllib3==1.26.5
Wand==0.6.10
wasabi==0.9.0
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==2.0.1
widgetsnbextension==3.5.1
xxhash==2.0.2
yacs==0.1.8
yarl==1.7.2
zipp==3.6.0

**Comment by chris-opendata** [comment]
I realized that it was caused by sharing a param group in a dual-optimizer setting where one is a minimizer and the other is a maximizer. So, creating param group instances separately fixes the problem.
